src/ast/nodes/IfStatement.ts

```diff
@@ -45,33 +50,41 @@ export default class IfStatement extends StatementBase {
 	}
 
 	render(code: MagicString, options: RenderOptions) {
-		const testValue = this.test.getValue();
+		const testValue = this.hasUnknownTestValue ? UNKNOWN_VALUE : this.test.getValue();
 		if (
 			!this.context.treeshake ||
 			this.test.included ||
 			(testValue ? this.alternate !== null && this.alternate.included : this.consequent.included)
 		) {
-			super.render(code, options);
+			this.test.render(code, options);
+			if (this.consequent.included) {
+				this.consequent.render(code, options);
+			} else {
+				code.overwrite(this.consequent.start, this.consequent.end, '{}');
+			}
+			if (this.alternate !== null) {
+				if (this.alternate.included) {
+					this.alternate.render(code, options);
+				} else {
+					code.remove(this.consequent.end, this.alternate.end);
+				}
+			}
 		} else {
-			// if test is not included, it is impossible that alternate===null even though it is the retained branch
+			// if test is not included, then the if statement is included because
+			// there is exactly one included branch
 			const branchToRetain = testValue ? this.consequent : this.alternate;
 			code.remove(this.start, branchToRetain.start);
 			code.remove(branchToRetain.end, this.end);
 			branchToRetain.render(code, options);
```

Comment:
It might be nice to try simplify this render logic a little.

Perhaps flip the parity and put this case at the top of render to catch first, with the else as the top block, so it's easier to see the fallthrough?

```js
if (this.context.treeshake && !test.included && testValue ? this.alternate === null || !this.alternate.included : !consequent.included) {
  const branchToRetain = testValue ? this.consequent : this.alternate;
  // ...
} else {
}
```

Reply:
You are right, flipping the logic makes it look much nicer. I also discovered another unchecked case where the test was not included even though both branches were included. This is now handled by overwriting the test with the actual stringified test value, further simplifying the output.